desc/magnetic_fields.py

```diff
@@ -1248,21 +1272,68 @@ def field_line_integrate(
         and the negative toroidal angle for negative Bphi
     field : MagneticField
         source of magnetic field to integrate
-    params: dict
+    params: dict, optional
         parameters passed to field
     grid : Grid, optional
         Collocation points used to discretize source field.
     rtol, atol : float
         relative and absolute tolerances for ode integration
     maxstep : int
         maximum number of steps between different phis
+    min_step_size: float
+        minimum step size (in phi) that the integration can take. default is 1e-8
+    solver: diffrax.Solver
+        diffrax Solver object to use in integration,
+        defaults to Tsit5(), a RK45 explicit solver
+    terminating_event: fxn
+        Function which takes as input the state of the ODE solution at each timestep
+        and **kwargs, and outputs a bool which, if True, terminates the solve at that
+        timestep.
+
+        fxn must have signature of (state, **kwargs) -> Bool
+
+        If not given and one of bounds_R,Z,phi are given, will
+        default to a terminating event which ends integration
+        once R,Z,or phi exit the domain defined by the given bounds
+
+        NOTE: If the solve is terminated early, the output returned is still
+        length(phis), however all values from the step point the fxn evaluated
+        to True and on will be inf
+        state has attributes such as state.y (array of length 3 with current (R,phi,Z))
+        see diffrax documentation for more in-depth information
+    bounds_R: tuple
```

Comment:
is this use case common enough to have these options for it vs just asking the user to supply a `DiscreteTerminatingEvent` object?

Reply:
I think yes